### PR TITLE
Add dart Fix for TextSelectionThemeData

### PR DESCRIPTION
**Pull request by Piinks**
This adds a dart fix for the deprecated ThemeData.[cursorColor, textSelectionColor, textSelectionHandleColor, useTextSelectionTheme] and migrates them to the appropriate TextSelectionThemeData parameters.

Part of https://github.com/flutter/flutter/issues/74908

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/fix_data.yaml [hunk 1]
```diff
@@ -12,6 +12,126 @@
...
+              expression: 'TextSelectionThemeData(cursorColor: {% cursorColor %}, selectionColor: {% textSelectionColor %}, selectionHandleColor: {% textSelectionHandleColor %},)'
+              requiredIf: "textSelectionColor != '' && cursorColor != '' && textSelectionHandleColor != ''"
+          - kind: 'removeParameter'
+            name: 'textSelectionColor'
+          - kind: 'removeParameter'
+            name: 'cursorColor'
+          - kind: 'removeParameter'
+            name: 'textSelectionHandleColor'
+          - kind: 'removeParameter'
+            name: 'useTextSelectionTheme'
```
Comment (Piinks) [inline packages/flutter/lib/fix_data.yaml:39]: This transform works, but only without the `removeParameter - useTestSelectionTheme`, I need that parameter to be removed, but it breaks the fix. :(

File: packages/flutter/lib/fix_data.yaml [hunk 2]
```diff
@@ -12,6 +12,137 @@
 version: 1
 transforms:
 
+  # Changes made in https://github.com/flutter/flutter/pull/66482
+  - title: "Remove 'brightness'"
```
Comment (Piinks) [inline packages/flutter/lib/fix_data.yaml:16]:
```suggestion
  - title: "Migrate to 'TextSelectionThemeData'"
```
```suggestion
  - title: "Remove 'brightness'"
```

**Comment by Piinks** [comment]
So these fixes are only partially working, this is the result of applying these rules:
```dart
// Changes made in https://github.com/flutter/flutter/pull/66482
  ThemeData(textSelectionTheme: TextSelectionThemeData(selectionColor: Colors.red)); ✅
  ThemeData(textSelectionTheme: TextSelectionThemeData(cursorColor: Colors.red)); ✅
  ThemeData(textSelectionTheme: TextSelectionThemeData(selectionHandleColor: Colors.red)); ✅
  ThemeData(); ✅
  // These removed useTextSelectionTheme, but did not apply the rest of the transform
  ThemeData(textSelectionColor: Colors.red); ✅ / ❌
  ThemeData(cursorColor: Colors.red); ✅ / ❌
  ThemeData(textSelectionHandleColor: Colors.red); ✅ / ❌
// These had no effect  
ThemeData( ❌
    textSelectionColor: Colors.red,
    cursorColor: Colors.red,
  );
  ThemeData( ❌
    textSelectionHandleColor: Colors.red,
    cursorColor: Colors.red,
  );
  ThemeData( ❌
    textSelectionColor: Colors.red,
    textSelectionHandleColor: Colors.red,
  );
  ThemeData( ❌
    textSelectionColor: Colors.red,
    cursorColor: Colors.red,
    textSelectionHandleColor: Colors.red,
  );
// Again, this removed useTextSelectionTheme, but did not apply the rest of the transform
  ThemeData( ✅ / ❌
    textSelectionColor: Colors.red,
    cursorColor: Colors.red,
    textSelectionHandleColor: Colors.red,
  );
```

**Comment by Piinks** [comment]
Getting closer, the cases where there is more than one of the color parameters in play is where it breaks. I am checking to see if any of the conditions are conflicting with each other
Current result of transforms:
```dart
// Changes made in https://github.com/flutter/flutter/pull/66482
  ThemeData(textSelectionTheme: TextSelectionThemeData(selectionColor: Colors.red));
  ThemeData(textSelectionTheme: TextSelectionThemeData(cursorColor: Colors.red));
  ThemeData(textSelectionTheme: TextSelectionThemeData(selectionHandleColor: Colors.red));
  ThemeData();
  ThemeData(textSelectionTheme: TextSelectionThemeData(selectionColor: Colors.red));
  ThemeData(textSelectionTheme: TextSelectionThemeData(cursorColor: Colors.red));
  ThemeData(textSelectionTheme: TextSelectionThemeData(selectionHandleColor: Colors.red));
  ThemeData(
    textSelectionColor: Colors.red,
    cursorColor: Colors.red,
  );
  ThemeData(
    textSelectionHandleColor: Colors.red,
    cursorColor: Colors.red,
  );
  ThemeData(
    textSelectionColor: Colors.red,
    textSelectionHandleColor: Colors.red,
  );
  ThemeData(
    textSelectionColor: Colors.red,
    cursorColor: Colors.red,
    textSelectionHandleColor: Colors.red,
  );
  ThemeData(
    textSelectionColor: Colors.red,
    cursorColor: Colors.red,
    textSelectionHandleColor: Colors.red,
    useTextSelectionTheme: false,
  );
```

**Comment by Piinks** [comment]
Currently blocked on https://github.com/dart-lang/sdk/issues/45439

**Review by goderbauer — APPROVED** [review]
LGTM

**Pull request merged**